pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/SortOperator.java

```diff
@@ -164,6 +164,9 @@ private void consumeInputBlocks() {
               _rows.addAll(container.subList(0, _numRowsToKeep - numRows));
               LOGGER.debug("Early terminate at SortOperator - operatorId={}, opChainId={}", _operatorId,
                   _context.getId());
+              setEarlyTerminate();
+              // acquire extra metadata block
+              block = _upstreamOperator.nextBlock();
```

Comment:
This is not very robust. Simply remove the `break` and let it follow the regular execution flow